lib/cn/move.c

```diff
@@ -29,54 +29,55 @@ cn_move(
     bool                       src_del,
     struct cn_tree_node       *tgt_node)
 {
-    struct cn_tree *tree;
-    struct kvset_list_entry *src, *src_end, *tgt;
+    struct kvset_list_entry *src, *tgt;
     struct list_head *src_head, *tgt_head;
-    uint64_t *src_ksidv = NULL;
+    struct cn_tree *tree;
     uint64_t src_nodeid;
-    merr_t err;
 
     INVARIANT(w && src_node && tgt_node);
-    INVARIANT((src_cnt > 0 && src_list) || (src_cnt == 0 && !src_list));
+    assert((src_cnt > 0 && src_list) || (src_cnt == 0 && !src_list));
```

Comment:
how about an "if and only if" logic:
```
assert( (src_cnt == 0) == (src_list == NULL) );
```

Reply:
or better:
```
assert( src == 0 == src_list == NULL );
```
:)

Reply:
Oops, I meant to change that back to INVARIANT.
How about this:  assert( !src_cnt == !src_list);   hahahahaha!